### PR TITLE
Add dynamic configuration

### LINKED ISSUE #45: Dynamic key configuration
Hello,

we need to allow users of our application to define at least `publicKey` and `privateKey` in settings. We can not hard-code them to `config.yml`.

I did not find any correct way to do this. Always some private members/methods were blocking the way.

Will you accept some PR to allow this dynamic setting? Will you give me some hint how you would implement that, so that I can implement this functionality in the way you like?

### PR DESCRIPTION
**Pull request by cuchac**
As discussed in #45 

Added default values to make similar DX as using symfony config - without specifying all config

### CODE REVIEW

**Comment by cuchac** [comment]
Another way to avoid specifying all the default values can be to simply add `getConfig()` and setters to config values. This way defaults would be in single place and it will be easier to use - just call `getConfig('default')->setPrivateKey('....')`.
Using current form one has to use `new ConfigDTO(.....)`, `addConfig()` and `setConfig()`.

File: src/LuigisBoxBundle/DTO/ConfigDTO.php [hunk 1]
```diff
@@ -44,13 +44,13 @@ class ConfigDTO
...
         string $privateKey,
-        float $connectionTimeout,
-        float $requestTimeout,
-        float $searchTimeout,
-        int $searchCacheTtl
+        string $host = 'https://live.luigisbox.com',
+        float $connectionTimeout = 4.0,
+        float $requestTimeout = 10.0,
+        float $searchTimeout = 6.0,
+        int $searchCacheTtl = 0
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/DTO/ConfigDTO.php]: please make const from `Configuration` class as public and use them here

File: tests/Acceptance/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -46,13 +50,43 @@ public function invalid(array $config, ?string $expectedMessage = null): void
...
+     * @test
+     */
+    public function addConfigWithColidingName()
+    {
+        $this->expectExceptionMessage('Configuration with key "config_name" already exists.');
+        $configProvider = ExampleConfiguration::provideDefaultConfig();
+
+        $config = new ConfigDTO('public', 'private');
+        $configProvider->addConfig('config_name', $config);
+    }
```
Comment (lukasz-falda) [inline tests/Acceptance/DependencyInjection/ConfigurationTest.php]: ConfigurationTest is a test for dependencyInjection configuration.
Maybe you can move this tests for another class ex. `\Answear\LuigisBoxBundle\Tests\Unit\DTO\ConfigDTOTest`?

**Comment by cuchac** [comment]
Done the refactoring

File: tests/Acceptance/DependencyInjection/ConfigurationTest.php [hunk 3]
```diff
@@ -34,9 +35,10 @@ public function validTest(array $configs, string $expectedConfigName): void
 
     /**
      * @test
+     *
      * @dataProvider provideInvalidConfig
      */
-    public function invalid(array $config, ?string $expectedMessage = null): void
+    public function invalid(array $config, string $expectedMessage = null): void
```
Comment (lukasz-falda) [inline tests/Acceptance/DependencyInjection/ConfigurationTest.php]: we are used to leaving `?` for nullable values

**Review by lukasz-falda — APPROVED** [review]

**Review by Wiktor6 — APPROVED** [review]

File: tests/Acceptance/DependencyInjection/ConfigurationTest.php (hunk 3 above)
Comment (cuchac) [reply tests/Acceptance/DependencyInjection/ConfigurationTest.php]: This is not my change. Tell this to your CS fixer :-)

Comment (lukasz-falda) [reply tests/Acceptance/DependencyInjection/ConfigurationTest.php]: 🙈 please, revert this, skip checking by adding `-n` on commiting. I will fix CS fixer then. Thanks

Comment (lukasz-falda) [reply tests/Acceptance/DependencyInjection/ConfigurationTest.php]: https://github.com/answear/luigis-box-bundle/pull/47 - merged fix for this

Comment (cuchac) [reply tests/Acceptance/DependencyInjection/ConfigurationTest.php]: done

**Pull request merged**